include/vcpkg/base/optional.h

```diff
@@ -266,151 +256,142 @@ namespace vcpkg
                 return *m_t;
             }
 
+            void destroy() { m_t = nullptr; }
+
         private:
             const T* m_t;
         };
     }
 
     template<class T>
-    struct Optional
+    struct Optional : private details::OptionalStorage<T>
     {
-    private:
-        details::OptionalStorage<T> m_base;
-
     public:
         constexpr Optional() noexcept { }
 
         // Constructors are intentionally implicit
         constexpr Optional(NullOpt) { }
 
-        template<class U, class = std::enable_if_t<!std::is_same_v<std::decay_t<U>, Optional>>>
-        constexpr Optional(U&& t) : m_base(std::forward<U>(t))
+        template<class U,
+                 std::enable_if_t<!std::is_same_v<std::decay_t<U>, Optional> &&
+                                      std::is_constructible_v<details::OptionalStorage<T>, U>,
+                                  int> = 0>
+        constexpr Optional(U&& t) : details::OptionalStorage<T>(static_cast<U&&>(t))
```

Comment:
Are we just going to not do std::forward anywhere now?

Reply:
I wouldn't call it out in a code review, but I broadly feel `std::forward` doesn't add to readability while injecting more abstraction, so I personally prefer `static_cast<T&&>`.